packages/vega-parser/src/parsers/guides/axis-util.js

```diff
@@ -0,0 +1,42 @@
+import { stringValue } from "vega-util";
+import { Top, Bottom, Left } from "./constants";
+
+export function xAxisExpr(signalExpr, ifXAxis, otherwise, ifXAxisIsString = true, otherwiseIsString = true) {
```

Comment:
- This works for both x and y, so the name should be `xy...`


- It returns a signal ref, so naming it `Expr` is misleading.

Reply:
Also, I found the flags `ifXAxisIsString = true, otherwiseIsString = true` hard to follow. 

I'd rather get rid of it and just apply stringValue when I call these, to simplify.

Reply:
It also seems like `signalExpr` is not an arbitrary `signalExpr`, but rather `orientSignal`.  Better give a specific name.

Reply:
The flag `ifXAxisIsString = true, otherwiseIsString = true` makes this method a bit harder to read when called. 

An alternative would be that `ifXAxis` can be either a string or a signalRef.  For the latter, no need to apply stringValue.

Reply:
refactored to discriminate signalRefs 👍